fragments/master.sh

```diff
@@ -211,9 +211,55 @@ start_k8s(){
 
 }
 
+pull_addons_image(){
+
+  docker pull \
+      fest/addons_services
+}
+
+run_addons_container(){
+
+  docker run \
+      --net=host \
+      -d \
+      fest/addons_services:latest
+}
+
+
+set +e
+poll_api_server(){
+
+  url='http://localhost:8080/healthz'
+  response=$(curl --write-out %{http_code} --silent --output /dev/null $url )
+  ok_status=200
+  timeout=1800 # 30 minutes
+  interval=5 # every 5 seconds
+
+  NEXT_WAIT_TIME=0
+  until [ $response -eq $ok_status ] || [ $NEXT_WAIT_TIME -eq $timeout ]; do
+    sleep $interval
+    response=$(curl --write-out %{http_code} --silent --output /dev/null $url )
+    let NEXT_WAIT_TIME=NEXT_WAIT_TIME+interval
+  done
+
+  if [ $response -eq $ok_status ]; then
+        run_addons_container
+  elif [ $NEXT_WAIT_TIME -eq $timeout ]; then
+        echo "Timeout has occurred while trying to contact the API service ..."
+  else
+        echo "Api service cannot be reched ..."
+  fi
+}
+set -e
+
 echo "Detecting your OS distro ..."
 detect_lsb
 
+set +e
+pull_addons_image &
```

Comment:
hmm..why not:

-Start api server
- wait until ready
- deploy add-ons

Makes sense?